src/server/scripts/EasternKingdoms/zone_tirisfal_glades.cpp

```diff
@@ -38,134 +36,161 @@ EndContentData */
 
 enum Calvin
 {
-    SAY_COMPLETE        = 0,
-    SPELL_DRINK         = 2639,                             // possibly not correct spell (but iconId is correct)
-    QUEST_590           = 590
+    SAY_COMPLETE = 0,
+    SPELL_FOOD = 7737,
+    QUEST_590 = 590
+};
+
+enum event
+{
+    EVENT_EMOTE = 1,
+    EVENT_TALK = 2,
+    EVENT_DRINK_AND_QUEST_COMQUEST = 3,
+    EVENT_END = 4,
 };
 
 class npc_calvin_montague : public CreatureScript
 {
 public:
-    npc_calvin_montague() : CreatureScript("npc_calvin_montague") { }
+    npc_calvin_montague() : CreatureScript("npc_calvin_montague") {}
 
     bool OnQuestAccept(Player* player, Creature* creature, Quest const* quest) override
     {
         if (quest->GetQuestId() == QUEST_590)
         {
-            creature->SetFaction(FACTION_ENEMY);
-            creature->SetImmuneToPC(false);
-            CAST_AI(npc_calvin_montague::npc_calvin_montagueAI, creature->AI())->AttackStart(player);
+            creature->SetFaction(FACTION_MONSTER);
+            creature->RemoveNpcFlag(UNIT_NPC_FLAG_QUESTGIVER);
+            creature->RemoveUnitFlag(UNIT_FLAG_IMMUNE_TO_PC);//This is reset when out of combat
+            creature->SetUnitFlag(UNIT_FLAG_PET_IN_COMBAT);
+            creature->SetReactState(REACT_AGGRESSIVE);
+            creature->AI()->AttackStart(player);
         }
         return true;
     }
 
-    CreatureAI* GetAI(Creature* creature) const override
-    {
-        return new npc_calvin_montagueAI(creature);
-    }
-
     struct npc_calvin_montagueAI : public ScriptedAI
     {
-        npc_calvin_montagueAI(Creature* creature) : ScriptedAI(creature) { }
-
-        uint32 m_uiPhase;
-        uint32 m_uiPhaseTimer;
-        ObjectGuid m_uiPlayerGUID;
+        npc_calvin_montagueAI(Creature* creature) : ScriptedAI(creature) {}
 
         void Reset() override
         {
-            m_uiPhase = 0;
-            m_uiPhaseTimer = 5000;
-            m_uiPlayerGUID.Clear();
-
-            me->RestoreFaction();
-
-            if (!me->IsImmuneToPC())
-                me->SetImmuneToPC(true);
+            if (m_uiPhase != 1)
+            {
+                me->SetReactState(REACT_DEFENSIVE);
+                me->RemoveUnitFlag(UNIT_FLAG_IMMUNE_TO_PC);
+            }
         }
 
-        void JustEngagedWith(Unit* /*who*/) override { }
-
-        void AttackedBy(Unit* pAttacker) override
+        void JustReachedHome() override
         {
-            if (me->GetVictim() || me->IsFriendlyTo(pAttacker))
-                return;
+            if (m_uiPhase == 1)
+            {
+                events.ScheduleEvent(EVENT_EMOTE, 2500ms);
+            }
+            else
+            {
+                events.ScheduleEvent(EVENT_END, 3min);
+            }
```

Comment:
just remove the brackets

Reply:
I'm sorry, I don't quite understand what this means
 `if (m_uiPhase == 1)`
Modify to
 `if (m_uiPhase)`
？？？

Reply:
```cpp
            if (m_uiPhase)
                events.ScheduleEvent(EVENT_EMOTE, 2500ms);
            else
                events.ScheduleEvent(EVENT_END, 3min);